src/components/progress-bar/progress-bar.scss

```diff
@@ -0,0 +1,205 @@
+  height: $progress-bar-height;
+  overflow: hidden;
+  position: relative;
+  /** translateZ is added to force the md-progress-bar into its own GPU layer. */
+  transform: translateZ(0);
+  transition: opacity $progress-bar-piece-animation-duration linear;
+  width: 100%;
+
+  .background {
+    background: url($buffer-bubbles-primary);
+    background-repeat: repeat-x;
+    background-size: 10px 4px;
+    height: 100%;
+    position: absolute;
+    visibility: hidden;
+    width: 100%;
+  }
+
+  .buffer {
+    background-color: md-color($md-primary, 100);
+    height: 100%;
+    position: absolute;
+    transform-origin: top left;
+    transition: transform $progress-bar-piece-animation-duration ease;
+    width: 100%;
+  }
+
+  .bar {
+    animation: none;
+    height: 100%;
+    position: absolute;
+    transform-origin: top left;
+    transition: transform $progress-bar-piece-animation-duration ease;
+    width: 100%;
+  }
+
+  .bar::after {
+    animation: none;
+    background-color: md-color($md-primary, 600);
+    content: '';
+    display: inline-block;
+    height: 100%;
+    position: absolute;
+    width: 100%;
+  }
+
+  .secondary {
+    visibility: hidden;
+  }
+
+  &.md-accent {
+    .background {
+      background: url($buffer-bubbles-accent);
+      background-repeat: repeat-x;
+      background-size: 10px 4px;
+    }
+    .buffer {
+      background-color: md-color($md-accent, 100);
+    }
+    .bar::after {
+      background-color: md-color($md-accent, 600);
+    }
+  }
+
+  &.md-warn {
+    .background {
+      background: url($buffer-bubbles-warn);
+      background-repeat: repeat-x;
+      background-size: 10px 4px;
+    }
+    .buffer {
+      background-color: md-color($md-warn, 100);
+    }
+    .bar::after {
+      background-color: md-color($md-warn, 600);
+    }
+  }
+
+  &[mode="query"] {
+    transform: rotateZ(180deg);
+  }
+
+  &[mode="indeterminate"],
+  &[mode="query"] {
+    .bar {
+      transition: none;
+    }
+    .primary {
+      animation: primary-indeterminate-translate $progress-bar-full-animation-duration infinite linear;
+      left: -145.166611%;
+    }
+    .primary.bar::after {
+      animation: primary-indeterminate-scale $progress-bar-full-animation-duration infinite linear;
+    }
+    .secondary {
+      animation: secondary-indeterminate-translate $progress-bar-full-animation-duration infinite linear;
+      left: -54.888891%;
+      visibility: visible;
+    }
+    .secondary.bar::after {
+      animation: secondary-indeterminate-scale $progress-bar-full-animation-duration infinite linear;
+    }
+  }
+
+  &[mode="buffer"] {
+    .background {
+      animation: background-scroll $progress-bar-piece-animation-duration infinite linear;
+      visibility: visible;
+    }
+  }
+}
+
+
+// Reverse the apparent directionality of progress vars for rtl.
+:host-context([dir="rtl"]) {
+  transform: rotateY(180deg);
+}
+
+
+/** Animations for indeterminate and query mode. */
```

Comment:
Could you point to the resource you used to get those percentage for animations? They seem like magic numbers without context.